perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+        custom_core_force.addGlobalParameter('lambda_bonds', 0.0)
+        custom_core_force.addPerBondParameter('length1') # old bond length
+        custom_core_force.addPerBondParameter('K1') # old spring constant
+        custom_core_force.addPerBondParameter('length2') # new bond length
+        custom_core_force.addPerBondParameter('K2') #new spring constant
+
+        self._hybrid_system.addForce(custom_core_force)
+        self._hybrid_system_forces['core_bond_force'] = custom_core_force
+
+        #add a bond force for environment and unique atoms (bonds are never scaled for these):
+        standard_bond_force = openmm.HarmonicBondForce()
+        self._hybrid_system.addForce(standard_bond_force)
+        self._hybrid_system_forces['standard_bond_force'] = standard_bond_force
+
+    def _add_angle_force_terms(self):
+        """
+        This function adds the appropriate angle force terms to the hybrid system. It does not add particles
+        or parameters to the force; this is done elsewhere.
+        """
+        energy_expression  = '(K/2)*(theta-theta0)^2;'
+        energy_expression += 'K = (1.0-lambda_angles)*K_1 + lambda_angles*K_2;' # linearly interpolate spring constant
+        energy_expression += 'theta0 = (1.0-lambda_angles)*theta0_1 + lambda_angles*theta0_2;' # linearly interpolate equilibrium angle
+
+        #create the force and add relevant parameters
+        custom_core_force = openmm.CustomAngleForce(energy_expression)
+        custom_core_force.addGlobalParameter('lambda_angles', 0.0)
+        custom_core_force.addPerAngleParameter('theta0_1') # molecule1 equilibrium angle
+        custom_core_force.addPerAngleParameter('K_1') # molecule1 spring constant
+        custom_core_force.addPerAngleParameter('theta0_2') # molecule2 equilibrium angle
+        custom_core_force.addPerAngleParameter('K_2') # molecule2 spring constant
+
+        #add the force to the system and the force dict.
+        self._hybrid_system.addForce(custom_core_force)
+        self._hybrid_system_forces['core_angle_force'] = custom_core_force
+
+        #add an angle term for environment/unique interactions--these are never scaled
+        standard_angle_force = openmm.HarmonicAngleForce()
+        self._hybrid_system.addForce(standard_angle_force)
+        self._hybrid_system_forces['standard_angle_force'] = standard_angle_force
+
+    def _add_torsion_force_terms(self):
+        """
+        This function adds the appropriate PeriodicTorsionForce terms to the system. Core torsions are interpolated,
+        while environment and unique torsions are always on.
+        """
+        energy_expression  = '(1-lambda_torsions)*U1 + lambda_torsions*U2;'
+        energy_expression += 'U1 = K1*(1+cos(periodicity1*theta-phase1));'
+        energy_expression += 'U2 = K2*(1+cos(periodicity2*theta-phase2));'
+
+        #create the force and add the relevant parameters
+        custom_core_force = openmm.CustomTorsionForce(energy_expression)
```

Comment:
Good point. For small molecules and GAFF, I _think_ that doesn't happen, if I recall the paper correctly. But in general it will so we should deal with it.